cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -811,6 +811,19 @@ impl TryFrom<PackageInfoCommon> for SearchResult {
     }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub enum ParsedSearch {
+    Clean(String),
+    VersionStripped(String),
+}
+
+pub fn parse_search_term(search_term: &str) -> ParsedSearch {
```

Comment:
> I think the version parsing has some overlap with https://github.com/flox/flox/pull/1562 though it should be easy to rebase in either direction.

@ysndr do you think it makes sense to combine them? `flox search` is currently more permissive in that it will allow invalid version specifiers, warn about them, and continue the search with them omitted.

Reply:
Yeah. My feeling is that this is different and simple enough that it's OK for the two implemntations to be separate. We'd want to revsit that if/when we implement searching by version in the future.

Reply:
a different nit would be to impl `parse_search_term` as a constructor of `ParsedSearch	` / `SearchTerm` e.g.

```
impl ParsedSearch {
	fn from_arg(search_term: &str) -> Self {}
}
```